### PR TITLE
sql/catalog/resolver: add a benchmark for ResolveFunction

### PR DESCRIPTION
**Pull request by ajwerner**
Epic: None

Release note: None

### CODE REVIEW

File: pkg/sql/catalog/resolver/resolver_bench_test.go [hunk 1]
```diff
@@ -113,3 +114,88 @@ func BenchmarkResolveExistingObject(b *testing.B) {
...
+			setup: []string{
+				"CREATE SCHEMA sc",
+				"CREATE FUNCTION sc.foo() RETURNS int IMMUTABLE LANGUAGE SQL AS $$ SELECT 1 $$",
+			},
+			name:       tree.MakeUnresolvedName("foo"),
+			flags:      tree.ObjectLookupFlagsWithRequired(),
+			searchPath: "public,$user,sc",
+		},
+	} {
+		b.Run(strings.Join(tc.setup, ";")+tc.name.String(), func(b *testing.B) {
```
Comment (chengxiong-ruan) [inline pkg/sql/catalog/resolver/resolver_bench_test.go:154]: the test name will probably be fun. probably manually give them some more readable names.

File: pkg/sql/catalog/resolver/resolver_bench_test.go [hunk 2]
```diff
@@ -113,3 +114,88 @@ func BenchmarkResolveExistingObject(b *testing.B) {
...
+				sp = &ec.SessionData().SearchPath
+			}
+
+			rs := p.(resolver.SchemaResolver)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				fd, err := rs.ResolveFunction(ctx, &tc.name, sp)
+				require.NoError(b, err)
+				require.NotNil(b, fd)
+			}
```
Comment (chengxiong-ruan) [inline pkg/sql/catalog/resolver/resolver_bench_test.go:198]: b.StopTimer()?

**Review by chengxiong-ruan — APPROVED** [review]
much appreciated!!

**Review by ajwerner — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89422#-:-NE6jsT5A0Hb4alwR1TR:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @chengxiong-ruan)

___
*[`pkg/sql/catalog/resolver/resolver_bench_test.go` line 154 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/89422#-NDdUMy9LPr9C4RImPXA-r1-154:-NE6jq_t1mIl26DncAOb:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d93ec59fa11445234e4ad2c6a2099bcb00e10529/pkg/sql/catalog/resolver/resolver_bench_test.go#L154)):*
<details><summary><i>Previously, chengxiong-ruan (Chengxiong Ruan) wrote…</i></summary><blockquote>

the test name will probably be fun. probably manually give them some more readable names.
</blockquote></details>

Done.
___
*[`pkg/sql/catalog/resolver/resolver_bench_test.go` line 198 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/89422#-NDdUMy9LPr9C4RImPXA-r1-198:-NE6jrfMDRhZShmZru_y:b-s3hmk9) ([raw file](https://github.com/cockroachdb/cockroach/blob/d93ec59fa11445234e4ad2c6a2099bcb00e10529/pkg/sql/catalog/resolver/resolver_bench_test.go#L198)):*
<details><summary><i>Previously, chengxiong-ruan (Chengxiong Ruan) wrote…</i></summary><blockquote>

b.StopTimer()?
</blockquote></details>

Not needed.

**Pull request merged**